### PR TITLE
`balances` - addresses and redesign

### LINKED ISSUE #102: avatars for wallet
#### What would you like to be added:
We have some pages were we need to use generic avatars. Proposal is to use https://github.com/boringdesigners/boring-avatars.
Colors: #FF00AF #00F3FF #F6911D and #F4F3F6 .

### LINKED ISSUE #463: view wallet on DeFiScan
#### What would you like to be added:
We can have a button in the wallet to view the current loaded wallet in DeFiScan. Something like `View on DeFiScan`. Not sure if we have that page yet.
#### Why is this needed:

/area ui-ux

Let me know your thoughts about it @fuxingloh @aentan  @jtlntno-cake

### LINKED ISSUE #1145: `header UI` - `RECEIVE` in german in top right of balance screen is wrapped
#### What UI/UX improvements would you like to be implemented: 
- Revisit if `RECEIVE` in balance header can be avoided to be wrapped into new line in german translation

#### Why is this needed:
- To maintain a cleaner header

#### Which component/screens does this concern?:
- Balance screen

#### Anything else we need to know?:
- Not sure if it is related to moving settings icon to top left of header
- Happens in samsung galaxy a52, not on iphone 12
image

### PR DESCRIPTION
**Pull request by thedoublejay**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
1. Redesigned Balances screen to display address and avatars, in-preparation for multi address functionality
2. Display avatar based on address
3. Move Receive button from header to Balance control card
4. Updated empty state for Balances
5. Tweaked colors for Eye icon

#### Which issue(s) does this PR fixes?:

Fixes #463 
Fixes #1145 
Fixes #102 

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [x] Unit tests*
- [x] Added e2e tests*
- [x] Added translations*

### CODE REVIEW

**Review by pierregee — DISMISSED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  return (
+    <ThemedView
+      testID='balance_control_card'
+      style={tailwind('p-4 flex flex-col')}
+      dark={tailwind('bg-gray-800 border-b border-gray-700')}
+      light={tailwind('bg-white border-b border-gray-100')}
+    >
+      <View style={tailwind('flex flex-row items-center')}>
+        <Avatar
+          size={30}
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx:29]: image
Figma uses size 40 tho, probably to distinguish with the token icon size (32) under the balances list

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                dark={tailwind('text-darkprimary-500')}
+                iconType='MaterialIcons'
+                light={tailwind('text-primary-500')}
+                name='open-in-new'
+                size={18}
+              />
+            </TouchableOpacity>
+          </View>
+          <TouchableOpacity onPress={async () => await openURL(getAddressUrl(address))}>
+            <ThemedText testID='wallet_address' style={tailwind('text-sm font-semibold pr-4')}>
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx:58]: image
The additional padding is leaving a wider gap on the right, is it required?

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+export function BalanceControlCard (): JSX.Element {
+  const { address } = useWalletContext()
+  const { getAddressUrl } = useDeFiScanContext()
+  const navigation = useNavigation<NavigationProp<BalanceParamList>>()
+  return (
+    <ThemedView
+      testID='balance_control_card'
+      style={tailwind('p-4 flex flex-col')}
+      dark={tailwind('bg-gray-800 border-b border-gray-700')}
+      light={tailwind('bg-white border-b border-gray-100')}
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx:25]: image
Can change the border color to a darker gray on light mode? Currently the border is same color as the background so can't distinguished, works fine on dark mode.

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx (hunk 2 above)
Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx:58]: Yep, too little text below looks like a floating letter even on smaller devices

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx (hunk 1 above)
Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx]: Updated

File: mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx (hunk 3 above)
Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/screens/Balances/components/BalanceControlCard.tsx]: Updated

**Review by kyleleow — APPROVED** [review]

**Pull request merged**